test/unit_tests/cellular/cellular.cpp

```diff
@@ -78,6 +99,64 @@ TEST_CASE("cellular_signal()") {
         }
     }
 
+    SECTION("CellularSignalHal RSSI and QUAL") {
+        NetStatus status;
+        memset(&status, 0, sizeof(status));
+        status.rssi = -50; // arbitrary non-zero values
+        status.qual = 37;  //  "
+
+        SECTION("CellularSignalHal::rssi and CellularSignalHal::qual are set to expected values") {
```

Comment:
I was told to use the BDD (Behavior Driven Development) style. SECTION (API subset) GIVEN (environment) WHEN (input) THEN (output).

I would have expected the structure to look more like this...

```
SECTION("CellularSignalHal RSSI and QUAL") {
  GIVEN("A known NetStatus structure") {
    WHEN("cellular_signal_impl is invoked with an uninitialized CellularSignalHal structure and known NetStatus structure") {
      THEN("the value of NetStatus::rssi is loaded into CellularSignalHal::rssi") {}
      THEN("the value of NetStatus::qual is loaded into CellularSignalHal::qual") {}
    }
  }
}
```

@m-mcgowan may have more input on this topic

Reply:
I followed the convention used in the existing file, they both mean the same thing.  I can change them if we feel it's necessary: https://github.com/catchorg/Catch2/blob/master/docs/test-cases-and-sections.md